tests/README.md

```diff
@@ -0,0 +1,120 @@
+# Honey-badger tests
+
+The `hbbft` crate comes with a toolkit for testing its various algorithms in simulated network environments.
+
+## Old vs new
+
+The old testing code can be found inside the `network` module and `.rs` files in the `tests` subdirectory that are not prefixed with `net_`. The newer networking code is contained inside the `net` module and the remaining `.rs` files.
+
+## VirtualNet
+
+Core of most tests is the `net::VirtualNet` struct, which simulates a network of nodes all running an instance of a distributed algorithm. Messages sent by these nodes are queued by the network. Every time the network is "cranked", a buffered message is delivered to its destination node and processed.
+
+Virtual networks can also host an adversary that can affect faulty nodes (which are tracked automatically) or reorder queued messages.
+
+To create a new network, the `NetBuilder` should be used:
+
+```rust
+// Create a network of 10 nodes, out of which 3 are faulty.
+let mut net = NetBuilder::new(0..10)
+    .num_faulty(3)
+    .using(move |node| { DynamicHoneyBadger::builder().build(node.netinfo) })
+    .build()
+    .expect("could not construct test network");
+```
+
+Algorithms that return a `Step` upon construction should use `using_step` instead.
+
+### Sending input
+
+`Input` can be sent to any node of the `VirtualNet` using the `send_input` method:
+
+```rust
+let input = ...;
+let _step = net.send_input(123, input).expect("algorithm failed");
+```
+
+While the resulting step is returned, it needn't be processed to keep the network going, as its messages are automatically added to the queue.
+
+Instead of targeting a node in particular, the same input can be sent to all nodes:
+
+```rust
+net.broadcast_input(input).expect("algorithm failed");
+```
+
+### Cranking the network
```

Comment:
I added those because I felt the word *crank* was not commonly used. I felt like we need an extra iteration-type of term, because we are using so many others in different areas already (iteration, step). The distinction really helps when talking about it.

Reply:
I will remove the quotes and try to reword the section a bit.

Reply:
I think that defining "crank" as the network handling a single message (I think... right? I may be wrong) would be beneficial here. As "crank" is a commonly used term in the code, but may not be totally understood without an explicit definition.